### PR TITLE
Radial Reynolds shear stress 

### PR DESCRIPTION
**Pull request by gyalla**
Adding the computation of the radial Reynolds shear stress and shear stress flux as actions to the reynoldsstress executor.

This assumes the flow is x-aligned, that we are operating on YZ planes, and that wake centers are defined with lateral and vertical centers. We should think of a way to generalize this for future work.

### CODE REVIEW

File: postproengine/reynoldsstress.py [hunk 1]
```diff
@@ -123,11 +142,74 @@ def execute(self, verbose=False):
...
+
+                Theta = np.arctan2(ZZ-z_center,YY-y_center)
+
+                self.parent.dbReAvg['uxur_avg'][iplane,:,:] = uv_avg * np.sin(Theta) + uw_avg * np.cos(Theta)
+                self.parent.dbReAvg['ux_avg_uxur_avg'][iplane,:,:] = ux_avg * self.parent.dbReAvg['uxur_avg'][iplane,:,:]
+
+            # Overwrite picklefile
+            if len(self.parent.pklfile)>0:
+                dbfile = open(self.parent.pklfile, 'wb')
+                pickle.dump(self.parent.dbReAvg, dbfile, protocol=2)
```
Comment (gyalla) [inline postproengine/reynoldsstress.py:271]: Is it OK to overwrite the parent pickle file if the user also wants to store uxur_avg and ux_avg_uxur_avg? Or should we store in different file?

File: postproengine/reynoldsstress.py [hunk 2]
```diff
@@ -123,11 +142,74 @@ def execute(self, verbose=False):
...
+                    y_center = yc
+                    z_center = zc
+
+                YY = self.parent.dbReAvg['y'][iplane,:,:]
+                ZZ = self.parent.dbReAvg['z'][iplane,:,:]
+                uv_avg  = self.parent.dbReAvg['uv_avg'][iplane,:,:]
+                uw_avg  = self.parent.dbReAvg['uw_avg'][iplane,:,:]
+                ux_avg  = self.parent.dbReAvg['velocityx_avg'][iplane,:,:]
+
+                Theta = np.arctan2(ZZ-z_center,YY-y_center)
```
Comment (gyalla) [inline postproengine/reynoldsstress.py:196]: Can you double check this part? Should match equation (22) in this reference: https://www.tandfonline.com/doi/abs/10.1080/14685248.2012.705005

File: postproengine/reynoldsstress.py (hunk 1 above)
Comment (lawrenceccheung) [reply postproengine/reynoldsstress.py:271]: yes, you can overwrite the parent dbfile if you're adding data to it (but not subtracting).

File: postproengine/reynoldsstress.py (hunk 2 above)
Comment (lawrenceccheung) [reply postproengine/reynoldsstress.py:196]: This looks correct.

**Review by lawrenceccheung — APPROVED** [review]

**Comment by gyalla** [comment]
To do: Add action to compute other Cartesian turbulent fluxes

**Comment by gyalla** [comment]
Added a further action to compute all Cartesian turbulent fluxes

File: postproengine/reynoldsstress.py [hunk 3]
```diff
@@ -123,11 +142,140 @@ def execute(self, verbose=False):
...
+
+                self.parent.dbReAvg['w_avg_uw_avg'][iplane,:,:] = w_avg * self.parent.dbReAvg['uw_avg'][iplane,:,:]
+                self.parent.dbReAvg['w_avg_vw_avg'][iplane,:,:] = w_avg * self.parent.dbReAvg['vw_avg'][iplane,:,:]
+                self.parent.dbReAvg['w_avg_ww_avg'][iplane,:,:] = w_avg * self.parent.dbReAvg['ww_avg'][iplane,:,:]
+
+                if include_radial:
+                    self.parent.dbReAvg['ux_avg_uxur_avg'][iplane,:,:] = u_avg * self.parent.dbReAvg['uxur_avg'][iplane,:,:]
+
+            # Overwrite picklefile
+            if len(self.parent.pklfile)>0:
```
Comment (gyalla) [inline postproengine/reynoldsstress.py:269]: Do we want to save all these to the same pickle file or is there a concern about the file getting too large? Should we separate turbulent stresses and fluxes?

**Pull request merged**